### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by veronicaburduzhan**
Thanks in advance for your review!

### CODE REVIEW

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  try {
+    const maxPrice = request.query.maxPrice;
+    const title = request.query.title;
+    const createdAfter = request.query.createdAfter;
+    const limit = request.query.limit;
+    let filteredMeals;
+    if (maxPrice) {
+      if (!isNaN(parseInt(maxPrice))) {
+        filteredMeals = meals.filter((meal) => meal.price < maxPrice);
+        response.send(filteredMeals);
```
Comment (Divyajg) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:16]: It is good idea to use filteredMeals 👍

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        response.send(400);
+      }
+    } else if (limit) {
+      if (!isNaN(parseInt(limit))) {
+        filteredMeals = meals.slice(0, limit);
+        response.send(filteredMeals);
+      } else {
+        response.send(400);
+      }
+    } else if (Object.keys(request.query).length === 0) {
```
Comment (Divyajg) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:43]: You have defined the queries as variables as maxPrice, limit, etc but not as object.
You need to define like this:
`const { maxPrice, title, createdAfter, limit } = request.query;`

File: nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+const router = express.Router();
+
+const reservations = require("../data/reservations.json");
+
+router.get("/", async (request, response) => {
+  response.send(reservations);
+});
+
+router.get("/:id", async (request, response) => {
+  response.send(reservations[request.params.id - 1]);
```
Comment (Divyajg) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js:11]: What if the first reservation id is 101 and the query parameter is 21? or
What if the reservation id 3 is deleted from the Database and the query parameter asked is 3?
Your logic will be failed.

File: nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+const router = express.Router();
+
+const reviews = require("../data/reviews.json");
+
+router.get("/", async (request, response) => {
+  response.send(reviews);
+});
+
+router.get("/:id", async (request, response) => {
+  response.send(reviews[request.params.id - 1]);
```
Comment (Divyajg) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js:11]: same as reservation by id

File: nodejs/week2/homework/warmup/index.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+app.use("/numbers", router);
+
+router.get("/add", (req, res) => {
+    const firstNumber = req.query.first;
+    const secondNumber = req.query.second;
+    res.send(`Sum is ${Number(firstNumber) + Number(secondNumber)}`);
+});
+
+router.get("/multiply/:firstNum/:secondNum", (req, res) => {
+  res.send(`Result is ${Number(req.params.firstNum) * Number(req.params.secondNum)}`);
```
Comment (Divyajg) [inline nodejs/week2/homework/warmup/index.js:14]: Good use of both query parameters and request query.

**Review by Divyajg — CHANGES_REQUESTED** [review]
Very good logic with meals request queries ⭐. 
Just a couple of changes requested.

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js (hunk 2 above)
Comment (veronicaburduzhan) [reply nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:43]: After this Sunday class, I felt that I'd made such mistakes. Thank you for specifying them for me:)

File: nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js (hunk 3 above)
Comment (veronicaburduzhan) [reply nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js:11]: Yes, you're absolutely right. Thanks!

File: nodejs/week2/homework/warmup/index.js (hunk 5 above)
Comment (veronicaburduzhan) [reply nodejs/week2/homework/warmup/index.js:14]: Thanks for your review. It'll help me with my homework for nodejs-week3 :)

**Pull request merged**